src/views/projects/View.svelte

```diff
@@ -48,23 +49,22 @@
     input: string,
     branches: Record<string, string>,
   ): { path?: string; revision?: string } {
+    const parsed: { path?: string; revision?: string } = {};
+    const commitPath = [input.slice(0, 40), input.slice(41)];
     const branch = Object.entries(branches).find(([branchName]) =>
       input.startsWith(branchName),
     );
-    const commitPath = [input.slice(0, 40), input.slice(41)];
-    const parsed: { path?: string; revision?: string } = {};
 
     if (branch) {
       const [rev, path] = [
         input.slice(0, branch[0].length),
         input.slice(branch[0].length + 1),
       ];
-
       parsed.revision = rev;
-      parsed.path = path ? path : "/";
+      parsed.path = path || "/";
```

Comment:
```diff
diff --git a/src/views/projects/View.svelte b/src/views/projects/View.svelte
index 59231789..c41456d3 100644
--- a/src/views/projects/View.svelte
+++ b/src/views/projects/View.svelte
@@ -61,10 +61,10 @@
         input.slice(branch[0].length + 1),
       ];
       parsed.revision = rev;
-      parsed.path = path || "/";
+      parsed.path = path ?? "/";
     } else if (utils.isOid(commitPath[0])) {
       parsed.revision = commitPath[0];
-      parsed.path = commitPath[1] || "/";
+      parsed.path = commitPath[1] ?? "/";
     } else {
       parsed.path = input;
     }

```

Reply:
Yeah this was also something I did first, but ended up using the `||` operator due to

`parsed.path` only ends up being `"/"` if `path` is either `null` or `undefined`.
If `path` is e.g. an empty string, (which can happen) we don't fallback to `"/"`.

The same happens with `commitPath[1]`.

We could be more explicit and set `path` to `undefined` or `null` explicitly if we found some edge cases like an empty string.

Reply:
Let's leave it as is then.